### ISSUE TITLE
Datetime parser rounds time to 24 hour

### ISSUE THREAD

**Issue by tkukushkin** [pymssql/pymssql#662]
Example code:
```python
import pymssql

conn = pymssql.connect("host.docker.internal", 'sa', 'QweAsdZxc!23', "tempdb")
cursor = conn.cursor()
cursor.execute("""
IF OBJECT_ID('persons', 'U') IS NOT NULL
    DROP TABLE persons
CREATE TABLE persons (
    Id INT NOT NULL PRIMARY KEY,
    RegisteredAt DATETIME2(7)
)
""")
cursor.execute("INSERT INTO persons VALUES (%s, %s)", (1, '2019-10-24 23:59:59.9990850'))
cursor.execute('SELECT * FROM persons')
row = cursor.fetchone()
conn.close()
```
Traceback:
```python
<ipython-input-2-5a3e882722dd> in <module>
     14 cursor.execute("INSERT INTO persons VALUES (%s, %s)", (1, '2019-10-24 23:59:59.9990850'))
     15 cursor.execute('SELECT * FROM persons')
---> 16 row = cursor.fetchone()
     17 conn.close()

src/pymssql.pyx in pymssql.Cursor.fetchone()

src/pymssql.pyx in pymssql.Cursor.getrow()

src/_mssql.pyx in _mssql.MSSQLRowIterator.__next__()

src/_mssql.pyx in _mssql.MSSQLConnection.fetch_next_row()

src/_mssql.pyx in _mssql.MSSQLConnection.get_row()

src/_mssql.pyx in _mssql.MSSQLConnection.convert_db_value()

ValueError: hour must be in 0..23
```
tsql -C
```
Compile-time settings (established with the "configure" script)
                            Version: freetds v1.00.104
             freetds.conf directory: /etc
     MS db-lib source compatibility: yes
        Sybase binary compatibility: no
                      Thread safety: yes
                      iconv library: yes
                        TDS version: auto
                              iODBC: no
                           unixodbc: yes
              SSPI "trusted" logins: no
                           Kerberos: no
                            OpenSSL: yes
                             GnuTLS: no
                               MARS: no
```
pymssql version: 2.1.4

**Comment by bemoody** [comment]
This seems to be a bug in the FreeTDS `dbconvert`/`dbdatecrack` functions:

    #include <stdio.h>
    #include <sybfront.h>
    
    int main()
    {
        DBDATETIMEALL dta;
        DBDATETIME dt;
        DBDATEREC dr;
        DBDATEREC2 dr2;
    
        dta.time = 86400ULL * 1000 * 1000 * 10 - 1;
        dta.date = 0;
    
        dbconvert(NULL, SYBMSDATETIME2, (void*) &dta, sizeof(dta),
                  SYBDATETIME, (void*) &dt, sizeof(dt));
        printf("dt: dtdays = %d, dttime = %d\n", dt.dtdays, dt.dttime);
    
        dbdatecrack(NULL, &dr, &dt);
        printf("dr: %04d-%02d-%02d %02d:%02d:%02d.%03d\n",
               dr.dateyear, dr.datemonth + 1, dr.datedmonth,
               dr.datehour, dr.dateminute, dr.datesecond, dr.datemsecond);
    
        dbanydatecrack(NULL, &dr2, SYBMSDATETIME2, &dta);
        printf("dr2: %04d-%02d-%02d %02d:%02d:%02d.%09d\n",
               dr2.dateyear, dr2.datemonth + 1, dr2.datedmonth,
               dr2.datehour, dr2.dateminute, dr2.datesecond, dr2.datensecond);
    }

displays:

    dt: dtdays = 0, dttime = 25920000
    dr: 1900-01-01 24:00:00.000
    dr2: 1900-01-01 23:59:59.999999900

whereas, presumably, `dttime` ought to always be less than 25920000 (86400 * 300).

**Comment by pauliev** [comment]
Confirmed with build 2.2.7 that this is still an issue but agree that looks like an issue with FreeTDS
```
CRITICAL hour must be in 0..23

  File "src/pymssql/_pymssql.pyx", line 521, in pymssql._pymssql.Cursor.fetchone
  File "src/pymssql/_pymssql.pyx", line 511, in pymssql._pymssql.Cursor.getrow
  File "src/pymssql/_mssql.pyx", line 470, in pymssql._mssql.MSSQLRowIterator.__next__
  File "src/pymssql/_mssql.pyx", line 1213, in pymssql._mssql.MSSQLConnection.fetch_next_row
  File "src/pymssql/_mssql.pyx", line 1384, in pymssql._mssql.MSSQLConnection.get_row
  File "src/pymssql/_mssql.pyx", line 875, in pymssql._mssql.MSSQLConnection.convert_db_value
```

**Comment by B-DAP** [comment]
is this still an issue to anyone? Any solutions to this problem that you would recommend?

**Comment by bemoody** [comment]
My recommendation is that pymssql (and other applications) should use
'dbanydatecrack' instead, because using SYBDATETIME as an intermediate
format will lose precision.

**Comment by termim** [comment]
Closed in bc2cc5040ba0453bd22798bd49ed4d39ed8df3e3
